packages/models-library/src/models_library/utils/common_validators.py

```diff
@@ -69,3 +71,36 @@ def null_or_none_str_to_none_validator(value: Any):
     if isinstance(value, str) and value.lower() in ("null", "none"):
         return None
     return value
+
+
+def create__check_only_one_is_set__root_validator(alternative_field_names: list[str]):
+    """Ensure exactly one and only one of the alternatives is set
+
+    NOTE: a field is considered here `unset` when it is `not None`. When None
+    is used to indicate something else, please do not use this validator.
+
+    This is useful when you want to give the client alternative
+    ways to set the same thing e.g. set the user by email or id or username
+    and each of those has a different field
+
+    NOTE: Alternatevely, the previous example can also be solved using a
+    single field as `user: Email | UserID | UserName`
+
+    SEE test_uid_or_email_are_set.py for more details
+    """
+
+    def _validator(cls, values):
+        assert set(alternative_field_names).issubset(cls.__fields__)  # nosec
+
+        got = {
+            field_name: values.get(field_name) for field_name in alternative_field_names
+        }
+
+        if not functools.reduce(operator.xor, (v is not None for v in got.values())):
+            msg = (
+                f"Either { 'or'.join(got.keys()) } must be set, but not both. Got {got}"
+            )
+            raise ValueError(msg)
+        return values
+
+    return _validator
```

Comment:
May I suggest to return directly an instance of root validator from here, given the same input fields?

I see that you are using it like so:

```python
    _check_uid_or_email = root_validator(allow_reuse=True)(
         create__check_only_one_is_set__root_validator(["uid", "email"])
     )
```

If you do something like below.

```suggestion
def check_only_one_is_set(alternative_field_names: list[str]):
    """Ensure exactly one and only one of the alternatives is set

    NOTE: a field is considered here `unset` when it is `not None`. When None
    is used to indicate something else, please do not use this validator.

    This is useful when you want to give the client alternative
    ways to set the same thing e.g. set the user by email or id or username
    and each of those has a different field

    NOTE: Alternatevely, the previous example can also be solved using a
    single field as `user: Email | UserID | UserName`

    SEE test_uid_or_email_are_set.py for more details
    """

    def _validator(cls, values):
        assert set(alternative_field_names).issubset(cls.__fields__)  # nosec

        got = {
            field_name: values.get(field_name) for field_name in alternative_field_names
        }

        if not functools.reduce(operator.xor, (v is not None for v in got.values())):
            msg = (
                f"Either { 'or'.join(got.keys()) } must be set, but not both. Got {got}"
            )
            raise ValueError(msg)
        return values

    return root_validator(allow_reuse=True)(_validator(alternative_field_names))
```

You can use it like:


```python
    _check_uid_or_email = check_only_one_is_set(["uid", "email"])
     )
```

Reply:
@GitHK I do not think it is a good idea to hide the type of validator. 
None of the other validators are used that way either.
Moreover, I do a static check inside to avoid misuse with the wrong validator.

Reply:
I somehow find it less readable this way. The function name did not help me in comprehending what was happening.

Reply:
@GitHK  please, check the manual https://docs.pydantic.dev/latest/concepts/validators/#reusing-validators